src/legacy/core_plugins/data/public/plugin.ts

```diff
@@ -87,22 +74,24 @@ export interface DataStart {
  * in the setup/start interfaces. The remaining items exported here are either types,
  * or static code.
  */
-export class DataPlugin
-  implements
-    Plugin<DataSetup, DataStart, DataPluginSetupDependencies, DataPluginStartDependencies> {
+
+export class DataPlugin implements Plugin<DataSetup, DataStart, {}, DataPluginStartDependencies> {
   private readonly indexPatterns: IndexPatternsService = new IndexPatternsService();
   private readonly query: QueryService = new QueryService();
   private readonly search: SearchService = new SearchService();
   private readonly timefilter: TimefilterService = new TimefilterService();
 
   private setupApi!: DataSetup;
+  private storage!: IStorageWrapper;
 
-  public setup(core: CoreSetup, { __LEGACY }: DataPluginSetupDependencies): DataSetup {
```

Comment:
I'm a little scared about the changes on the way the storage is now accessed. What I mean is that before, the storage access was provided by a setup dependency (`DataPluginSetupDependencies`), and now the storage can be accessed basically from anywhere in the code base with a simple `new Storage(window.localStorage)`, which may makes the migration more difficult when we'll implement https://github.com/elastic/kibana/issues/41982, as more calls may be added in places where there is no proper access to core APIs.

@joshdover WDYT ?

Reply:
@pgayvallet I agree that the way that storage was and is handled right now is not great. Even before this PR, apps would randomly create their own storage representation and pass it around. And this PR **doesn't change** the way this is handled (checkout `x-pack/legacy/plugins/graph/public/app.js` or `x-pack/legacy/plugins/lens/public/app_plugin/plugin.tsx` for example).

The `__LEGACY` wrapper was there only to mark the fact that `Storage` was imported from `ui/public` and not from NP. Now that it's moved to NP, we don't need that any more.

What we should do is, once platform team offers a core  storage access service, is replace every occurrence of `new Storage` with that service.